src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -6,19 +6,121 @@
  * За бажанням можна реалізувати інші методи інтрефейсу Map.</p>
  */
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private int size;
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final double LOAD_FACTOR = 0.75;
+    private Node<K, V>[] objects;
+
+    public MyHashMap() {
+        size = 0;
+        objects = new Node[DEFAULT_CAPACITY];
+    }
+
+    private void resize() {
+        int load = 0;
+        for (Node<K, V> object: objects) {
+            if (object != null) {
+                load++;
+            }
+        }
+        if (load >= objects.length * LOAD_FACTOR) {
+            Node<K, V>[] temporary = new Node[objects.length * 3 / 2];
+            temporary[0] = objects[0];
+            for (int i = 1; i < load; i++) {
+                if (objects[i] != null) {
+                    Node<K, V> moveNode = objects[i];
+                    temporary[moveNode.hash % temporary.length] = objects[i];
+                }
+            }
+            objects = temporary;
+        }
+    }
+
+    private int hash(K key) {
+        if (key == null) {
+            return 0;
+        }
+        int h = key.hashCode();
+        h = h ^ (h >>> 16);
+        if (h < 0) {
+            h = -1 * h + 1;
+        }
+        return (key == null) ? 0 : h;
+    }
 
     @Override
-    public void put(K key, V value) {
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || obj.getClass() != this.getClass()) {
+            return false;
+        }
+        K key = (K) obj;
 
+        return this.equals(obj);
+    }
+
+    @Override
+    public void put(K key, V value) {
+        resize();
+        Node<K, V> nodeToAdd = new Node<>(key, value);
+        nodeToAdd.hash = hash(key);
+        if (objects[nodeToAdd.hash % objects.length] == null) {
+            objects[nodeToAdd.hash % objects.length] = nodeToAdd;
+            nodeToAdd.next = null;
+            size++;
+        } else {
+            Node<K, V> inLinkObject = objects[nodeToAdd.hash % objects.length];
+            int count = 0;
+            while (inLinkObject != null) {
+                if (nodeToAdd.hash == inLinkObject.hash && (nodeToAdd.key == inLinkObject.key
+                        || nodeToAdd.key.equals(inLinkObject.key))) {
+                    inLinkObject.value = value;
+                    count++;
+                    break;
+                }
+                inLinkObject = inLinkObject.next;
+            }
+            if (count == 0) {
```

Comment:
try to simplify this method

Reply:
I tried to make it in one while() loop, but in this case or I'm not able to compare the last object in link, or I can't properly add new object to the link... So it looks like I did my best... even if it is not enough...